lib/spack/spack/test/sbang.py

```diff
@@ -201,16 +204,79 @@ def test_install_sbang(install_mockery):
     assert not os.path.exists(sbang_bin_dir)
 
     sbang.install_sbang()
-    check_sbang()
+    check_sbang_installation()
 
     # put an invalid file in for sbang
     fs.mkdirp(sbang_bin_dir)
     with open(sbang_path, "w") as f:
         f.write("foo")
 
     sbang.install_sbang()
-    check_sbang()
+    check_sbang_installation()
 
     # install again and make sure sbang is still fine
     sbang.install_sbang()
-    check_sbang()
+    check_sbang_installation()
+
+
+def test_sbang_fails_without_argument():
+    sbang = which(spack.paths.sbang_script)
+    sbang(fail_on_error=False)
+    assert sbang.returncode == 1
+
+
+def test_sbang_with_different_shebangs(tmpdir):
+    script = str(tmpdir.join("script"))
+
+    def check_sbang(shebang, returncode, expected=None):
+        # write a script out with <shebang> on second line
+        with open(script, "w") as f:
+            f.write("#!/bin/sh {sbang}\n{shebang}\n".format(
+                sbang=spack.paths.sbang_script,
+                shebang=shebang
+            ))
+        fs.set_executable(script)
+
+        # test running the script in debug, which prints what would be executed
+        exe = which(script)
+        out = exe(output=str, fail_on_error=False, env={"SBANG_DEBUG": "1"})
+
+        # check error status and output vs. expected
+        assert exe.returncode == returncode
+
+        if expected is not None:
+            assert expected == out.strip()
+
+    # perl, with and without /usr/bin/env
+    check_sbang("#!/path/to/perl", 0, "/path/to/perl -x %s" % script)
```

Comment:
(Comment not a request) I wonder if it would be a slight improvement to have this line:
```console
expected = expected_fmt % script
check_sbang(shebang, returncode, expected)
```
as the only test and parametrized fixtures to check each one of the cases below singularly, since they don't depend on each other.

Reply:
Good idea. Done!